Facades/CheckoutSession/CheckoutSessionFacade.php

```diff
@@ -0,0 +1,300 @@
+     * @param $basketSignature
+     * @param $currencyShortName
+     * @return CheckoutSession
+     * @throws \Doctrine\ORM\ORMException
+     * @throws \Doctrine\ORM\OptimisticLockException
+     * @throws \Enlight_Exception
+     * @throws \Mollie\Api\Exceptions\ApiException
+     */
+    public function startCheckoutSession($basketUserId, $paymentShortName, $basketSignature, $currencyShortName)
+    {
+        # immediately clear the restorable order
+        # just to make sure we don't have a previous one accidentally
+        $this->restorableOrder = null;
+
+        $basketData = $this->sBasket->sGetBasketData();
+
+        $this->logger->debug(
+            'Starting checkout for user: ' . $basketUserId . ' with payment: ' . $paymentShortName,
+            array(
+                'basket' => array(
+                    'amount' => $basketData['Amount'],
+                    'quantity' => $basketData['Quantity'],
+                    'payment' => $paymentShortName,
+                    'user' => $basketUserId
+                )
+            )
+        );
+
+        if (!$this->sBasket->sCountBasket()) {
+            throw new \Exception('No items in basket');
+        }
+
+
+        # build and create our transaction
+        # this is the most important line that helps us
+        # to get the bridge between the mollie payments and the shopware orders.
+        # it contains all necessary things for upcoming workflows.
+        $transaction = $this->buildTransaction($basketSignature, $currencyShortName);
+        $this->modelManager->persist($transaction);
+        $this->modelManager->flush();
+
+
+        if ($this->config->createOrderBeforePayment()) {
+
+            # create the order in our system.
+            # attention, this is the point where the basket is officially "empty"
+            # because the order is completed in Shopware
+            $orderNumber = $this->swOrderBuilder->createOrderBeforePayment(
+                $transaction->getTransactionId(),
+                $basketSignature
+            );
+
+            $order = $this->orderService->getShopwareOrderByNumber($orderNumber);
+
+            if (!$order instanceof Order) {
+                throw new \Exception('The order with order number ' . $orderNumber . ' could not be found');
+            }
+
+            $this->logger->debug('Created Order: ' . $orderNumber . ' for Transaction: ' . $transaction->getId());
+
+            # now that we have the order, we need to remember it
+            # this is required to restore the basket from that order in case of
+            # any additional failures below. otherwise the cart is empty because
+            # the order is already completed.
+            $this->restorableOrder = $order;
+
+            # now update all required references of our transaction
+            $transaction->setOrderId($order->getId());
+            $transaction->setOrderNumber($orderNumber);
+
+            # todo fehlt hier flush?
```

Comment:
todo found

Reply:
jap, checking that while testing today, thx